src/rard/research/tests/views/test_mentions.py

```diff
@@ -64,131 +116,263 @@ def test_default_queryset(self):
 
         # create some data and then show we don't find any of it
         # unless we add search terms to the request
-        Antiquarian.objects.create(name="findme", re_code="1")
-        Antiquarian.objects.create(name="andme", re_code="2")
+        Antiquarian.objects.create(name="findme", re_code="4")
+        Antiquarian.objects.create(name="andme", re_code="5")
 
         self.view.request = self.request(data={})
         self.assertEqual(0, len(self.view.get_queryset()))
 
-    def test_search_queryset(self):
+    def test_basic_search(self):
+        view = self.view
+        # antiquarians
+        self.assertEqual(len(list(view.basic_search(view, "aq", ""))), 3)
+        self.assertEqual(
+            list(view.basic_search(view, "aq", "an")), [self.andrew, self.antman]
+        )
 
-        # create some data to search
-        Antiquarian.objects.create(name="andrew", re_code="1")
-        Antiquarian.objects.create(name="antman", re_code="2")
-        Work.objects.create(name="andropov")
-        self.view.request = self.request(
+        # testimonia
+        self.assertEqual(len(list(view.basic_search(view, "tt", ""))), 3)
+        self.assertEqual(
+            list(view.basic_search(view, "tt", "an")), [self.tt2, self.tt1]
+        )
+
+        # topics
+        self.assertEqual(len(list(view.basic_search(view, "tp", ""))), 3)
+        self.assertEqual(
+            list(view.basic_search(view, "tp", "s")),
+            [self.pictures, self.coups, self.estates],
+        )
+
+    def test_bibliography_search(self):
```

Comment:
I can't see any examples here where you try to match the contents of bibliography titles. It'd be good to test for something that matches across both fields on different objects; e.g. "two" matching bi 2 and bi3 if the title for bi3 was "peppers two apples"